app/controllers/admin/users_controller.rb

```diff
@@ -23,12 +23,12 @@ def sort_direction
   end
 
   def restrict_access
-    redirect_to session[:last_moudle] unless current_user.
+    redirect_to current_user unless current_user.
```

Comment:
:/ this sounds .... weird and was kinda "hard to read", I would create a helper method called "autorized?" or something alike and refactor this as:
```
...
redirect_to(current_user) if authorized?
...

def authorized?
   current_user.is_admin || current_user.is_badges_admin?
end
```

even further, you could move the whole logic to the user ... I mean, reviewing the ```roles.include?``` at controller level sounds like knowing too much; the other way would be:
```
def restrict_access
  redirect_to current_user if current_user.is_bades_admin?
end
```